mpeg2/src/temi.rs

```diff
@@ -0,0 +1,282 @@
+use crate::{DecodeError, EncodeError};
+use byteorder::{BigEndian, ByteOrder};
+use core2::io::Write;
+
+pub const AF_DESCR_TAG_TIMELINE: u8 = 0x04;
+
+#[derive(Debug, PartialEq, PartialOrd, Copy, Clone)]
+#[allow(non_camel_case_types)]
+#[repr(C)]
+pub enum TimestampLength {
+    None = 0,
+    ThirtyTwoBits = 1,
+    SixtyFourBits = 2,
+}
+
+impl Default for TimestampLength {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct TEMITimelineDescriptor {
+    pub has_timestamp: TimestampLength,
+    pub timescale: u32,
+    pub media_timestamp: u64,
+
+    pub ntp_timestamp: Option<u64>,
+    pub ptp_timestamp: Option<u128>,
+
+    pub drop: bool,
+    pub frames_per_tc_seconds: u16,
+    pub duration: u16,
+    pub has_timecode: TimestampLength,
+    pub time_code: u64,
+
+    pub force_reload: bool,
+    pub paused: bool,
+    pub discontinuity: bool,
+    pub timeline_id: u8,
+}
+
+impl TEMITimelineDescriptor {
+    pub fn encode_len(&self) -> usize {
```

Comment:
In `AdaptationField`, this function is named "encoded_len". We should do the same here:

```suggestion
    pub fn encoded_len(&self) -> usize {
```

Reply:
👍